src/library/scala/collection/immutable/Vector.scala

```diff
@@ -544,6 +601,26 @@ class VectorIterator[+A](_startIndex: Int, endIndex: Int)
 
   private[this] var _hasNext = blockIndex + lo < endIndex
 
+  override def drop(n: Int): Iterator[A] = {
+    if (n > 0) {
+      lo = lo + n
```

Comment:
What should happen in case of overflows?

Reply:
need to guard for integer overflow. Collection overflow work Ok though

Reply:
guarded against integer overflow